src/Filter.tsx

```
@@ -49,192 +1,233 @@
+
+    const handleFilterTypeChange = (option: SelectableValue<string>) => {
+      let newFilter: GAFilter = {
+        ...filter,
+        filterType: option.value as GADimensionFilterType,
+      };
+
+      // Set initial values based on filter type
+      switch (option.value) {
+        case GADimensionFilterType.STRING:
+          newFilter.stringFilter = { matchType: GAStringFilterMatchType.EXACT, value: '', caseSensitive: false };
+          delete newFilter.inListFilter;
```

Biome

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)